qiskit/circuit/parameterexpression.py

```diff
@@ -490,6 +499,52 @@ def __eq__(self, other):
             return len(self.parameters) == 0 and complex(self._symbol_expr) == other
         return False
 
+    def __lt__(self, other):
+        """Check if this parameter expression is less than another parameter expression
+           or a fixed value (only if this is a bound expression).
+        Args:
+            other (ParameterExpression or a number):
+                Parameter expression or numeric constant used for comparison
+        Returns:
+            bool: result of the comparison
+        """
+        if isinstance(other, ParameterExpression):
+            if HAS_SYMENGINE:
+                from sympy import sympify
+
+                return sympify(self._symbol_expr).__lt__(sympify(other._symbol_expr))
+            else:
+                return self._symbol_expr.__lt__(other._symbol_expr)
+        elif isinstance(other, numbers.Number):
+            return len(self.parameters) == 0 and float(self._symbol_expr) < other
+        return False
+
+    def __gt__(self, other):
+        """Check if this parameter expression is greater than another parameter expression
+           or a fixed value (only if this is a bound expression).
+        Args:
+            other (ParameterExpression or a number):
```

Comment:
Use type annotations instead of adding the types to the docstring

Reply:
I tried to implement `__lt__` and `__gt__` similar to the already existing `__eq__` method. As this method is not implemented with type annotation, shall I also change this?

Reply:
You can just remove the type information from the docstring - we have Sphinx extensions that will automatically pull it in during the documentation build.  It's not generally a problem to have both, though - there are some cases where it's better to have a neater form specifically for Sphinx, just for legibility.